transport/wseb/src/main/java/org/kaazing/gateway/transport/wseb/WsebSession.java

```diff
@@ -393,12 +421,16 @@ public void attachReader(final HttpSession newReader) {
             if (ALIGN_UPSTREAM) {
                 final Thread ioThread = getIoThread();
                 final Executor ioExecutor = getIoExecutor();
+                // Prevent messageReceived from being fired from setIoAlignment and racing ahead of attachReader0
+                newReader.suspendRead();
```

Comment:
wow. What about other events like idle, closing etc ? Will they cause any problems ? For example, attach reader is schedule on a different thread, but underlying connection is closed.